uiServices/services_test.go

```diff
@@ -0,0 +1,207 @@
+package services
+
+import (
+	"testing"
+	"os"
+	"net/http/httptest"
+	"strings"
+	"github.com/go-chi/chi"
+	"io"
+	"net/http"
+	"github.com/Sirupsen/logrus"
+	"github.com/xtracdev/automated-perf-test/perfTestUtils"
+	"github.com/stretchr/testify/assert"
+)
+
+
+var (
+	reader  io.Reader
+)
+func TestFilePathExist(t *testing.T) {
+	path :=os.Getenv("GOPATH")+"/src/github.com/xtracdev/automated-perf-test/config/"
+	actual := false
+	actual = FilePathExist(path)
+	expected := true
+	assert.Equal (t, expected, actual)
+}
+
+func TestFilePathDoesNotExist(t *testing.T) {
+	path := "C:/github.com/xxx"
+	actual := FilePathExist(path)
+	expected := false
+	assert.Equal (t, expected, actual)
+}
+
+func TestValidJsonPost(t *testing.T) {
+	r := chi.NewRouter()
+	r.Mount("/", getIndexPage())
+	
+	configJson := `{
+        "apiName": "TestConfig",
+       "targetHost": "localhost",
+       "targetPort": "9191",
+       "numIterations": 1000,
+       "allowablePeakMemoryVariance": 30,
+       "allowableServiceResponseTimeVariance": 30,
+       "testCaseDir": "./definitions/testCases",
+       "testSuiteDir": "./definitions/testSuites",
+        "baseStatsOutputDir": "./envStats",
+       "reportOutputDir": "./report",
+       "concurrentUsers": 50,
+       "testSuite": "suiteFileName.xml",
+       "memoryEndpoint": "/alt/debug/vars",
+       "requestDelay": 5000,
+       "TPSFreq": 30,
+       "rampUsers": 5,
+       "rampDelay": 15
+       }`
+
+	reader= strings.NewReader(configJson)
+	r.HandleFunc("/configs", configsHandler)
+
+	filePath :=os.Getenv("GOPATH")+"/src/github.com/xtracdev/automated-perf-test/config/"
+
+	request, err := http.NewRequest(http.MethodPost, "/configs", reader)
+	request.Header.Set("configPathDir", filePath)
+
+	w := httptest.NewRecorder()
+	r.ServeHTTP(w, request)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	if http.StatusCreated != 201 {
```

Comment:
always use the http constants. and in the implementation, you're using statusOK. this test should check the status code is 201

at the minute, you're checking that http.StatusCreated (which is a constant) is 201. It always will be.

You need to check `if w.Code() == http.StatusCreated`